src/full/Agda/Interaction/Options/Base.hs

```diff
@@ -1656,6 +1685,15 @@ pragmaOptions = concat
   , pragmaFlag      "sized-types" lensOptSizedTypes
                     "enable sized types" "(inconsistent with --guardedness)"
                     $ Just "disable sized types"
+  , pragmaFlag      "type-based-termination" lensOptTypeBasedTermination
+                    "enable type-based termination checker" "(ignored if used together with --guardedness, --sized-types)"
+                    $ Just "disable type-based termination checker"
+  , pragmaFlag      "size-preservation" lensOptSizePreservation
+                    "enable size preservation inference during type-based termination checking" "(has effect only if --type-based-termination is enabled)"
```

Comment:
Also here: if the flag has no effect unless type-based-termination is enabled, then enabling it without that flag should be an error

Reply:
Looks like we have no precedent of statically conflicting flags. I'll make `--size-preservation` implying `--type-based-termination`.